### PR TITLE
Fix ClonesArguments for null and arrays

### PR DESCRIPTION
**Pull request by tombrown52**
The ClonesArguments class works for many cases, but fails when trying to clone an argument that is an array, or is null. This PR fixes those edge cases. This does not directly resolve any open issue, but it is related to #1153, #1469

check list

 - [x] Read the [contributing guide](https://github.com/mockito/mockito/blob/release/2.x/.github/CONTRIBUTING.md)
 - [x] PR should be motivated, i.e. what does it fix, why, and if relevant how
 - [x] If possible / relevant include an example in the description, that could help all readers
       including project members to get a better picture of the change
 - [x] Avoid other runtime dependencies
 - [x] Meaningful commit history ; intention is important please rebase your commit history so that each
       commit is meaningful and help the people that will explore a change in 2 years
 - [x] The pull request follows coding style
 - [x] Mention `Fixes #<issue number>` in the description _if relevant_
 - [x] At least one commit should mention `Fixes #<issue number>` _if relevant_

### CODE REVIEW

**Comment by tombrown52** [comment]
@TimvdLippe I believe I have the right branch this time. Please review

File: src/test/java/org/mockitousage/stubbing/CloningParameterTest.java [hunk 1]
```diff
@@ -26,12 +27,12 @@ public void shouldVerifyEvenIfArgumentsWereMutated() throws Exception {
         businessLogic(emailSender);
 
         // then
-        verify(emailSender).sendEmail(1, new Person("Wes"));
+        verify(emailSender).sendEmail(eq(1), eq(new Person("Wes")), (Person)any(), (Person[])any());
```
Comment (TimvdLippe) [inline src/test/java/org/mockitousage/stubbing/CloningParameterTest.java]: Could you revert these changes and instead add a new method with multiple arguments. (and thus add a new test to use that method separately)

**Comment by tombrown52** [comment]
@TimvdLippe I made new tests instead of extending the existing one. Please review again.
